DigitalLearningSolutions.Web/Controllers/RegisterController.cs

```diff
@@ -54,9 +55,15 @@ public IActionResult Index()
                     });
                 TempData.Set(delegateRegistrationData);
 
-                return View();
+                ViewBag.CentreOptions = SelectListHelper.MapOptionsToSelectListItemsWithSelectedValue
+                    (centresDataService.GetActiveCentresAlphabetical(), null);
+
+                return View(delegateRegistrationData.RegisterViewModel);

```

Comment:
Do we need this twice? Could we remove L28-61 potentially? As delegateRegistrationData.RegisterViewModel.Centre will be null and email validation passes with email being null. Now that I think about it, I think the validation only needs to happen on POST.

Reply:
The validation needs to happen on GET in the unlikely case that someone gets beyond this page of the form, then someone else finishes registering with the same email address, and the first person finishes registering and gets a duplicate email error so we boot them back to this page. But yeah Lines 58-61 can go for sure.

Reply:
Wait, how does a user get to the next page of the journey without posting the form (I'm not sure if I understood your description correctly)? 🤔

Reply:
Yeah I don't think I was clear enough. We need to have the check on a GET in case the registration stored procedure returns an error saying the email address is in use. There's only one case that I can think of where that might happen though:

1. A user completes the first page of the registration form
2. A different user (or the same user on a different computer) completes the whole registration form with the same centre and email address
3. The first user now tries to submit the form but the stored procedure gives us an error saying the email is already in use
4. We redirect the user back to the first page of the form which is a GET request, so to display the correct error we need to make the duplicate email check on a GET

Reply:
Oh I see - that makes sense! Could we add a quick comment about this here so if anyone ever reads through this, they won't get confused? :)